include/hffix.hpp

```diff
@@ -385,15 +391,16 @@ inline bool atotime(
  * w.push_back_string(hffix::tag::SendingTime, "20180309-13:46:01.0123456789123456");
  * \endcode
  */
-class message_writer {
+template <std::size_t BodyLengthDigits = 6>
+class basic_message_writer {
```

Comment:
I'm not sure about this, putting the template parameter directly on ``message_writer`` changes the API, since you must refer to it as ``message_writer<>``, so this avoids that at least.

Reply:
It's a great idea to set an explicit `BodyLengthDigits` and then throw if the use tries to write farther than the `BodyLengthDigits`  maximum size.

Introducing this `basic_message_writer` class does make the API not quite as totally simple-minded as it is now, though, and I like a really boringly obvious simple API. Also `basic_message_writer` complicates the Doxygen a lot.

How about we make `BodyLengthDigits` be a runtime member value `_body_length_digits` rather than a compile-time template parameter? The default constructor can initialize `_body_length_digits` to _6_, and we can make an overloaded constructor which accepts a `body_length_digits` parameter.